app/controllers/bookingsync_application/webhooks/base_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Webhooks::BaseController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :verify_signature
+
+  protected
+
+  def invalidate_request_and_log(message)
+    logger.info "Failing with 422 because: #{message}"
+    head 422
+  end
+
+  def verify_signature
+    Rails.logger.debug "bookingsync_hook_raw_post: #{request.raw_post}"
+    unless signature == request.headers["X-Content-Signature"]
+      invalidate_request_and_log "Bad BookingSync signature"
+    end
+  end
+
+  def signature
```

Comment:
lets start with the current way, and the other way should be introduced together with pr to core actually fixing this? but his should be tested in the dummy app, simple controller spec with correct and incorrect payload, pick some sample webhook from any app